With $M, r, y_0 > 0$, the logistic difference equation is defined, For $n \in \mathbb R$. 

as $y_n := y_n-_1 + ry_n-_1(M - y_n-_1).$

$M_b:=48000$

$y_0=1$ for $n\in \mathbb R$

$y_n$ is the predicted number of infected individuals in the $buffalo \ \  population$ after $n$ days

$Question \  4.1$

In [24]:
def logistic_non_recursive(n, r, M):
    K=[1]
    for v in range(1, n+1):
        K.append(K[v-1]+r*K[v-1]*(M-K[v-1]))
    
    return K[n]

In [25]:
M=48000

r_without_preventative_measures=2.1*10**(-6)

r_with_preventative_measures=6.7*10**(-7)

In [26]:
logistic_non_recursive(0,r_with_preventative_measures , M)

1

In [27]:
logistic_non_recursive(23615,r_with_preventative_measures , M)

47999.99999999989

$Question \ 4.2$

Calculating the first $n$ For $y_n ≥ 48000/2$ with preventative mearsures

In [28]:
n=0
M_buffalo=48000
r_with_preventative_measures=6.7*10**(-7)
k=48000/2
while logistic_non_recursive(n,r_with_preventative_measures,M_buffalo)<=k:
    n+=1

In [29]:
n

340

$Therefore\  n≥340$

$Question \ 4.3$

Calculating the first $n$ For $y_n ≥ 48000/2$ without preventative mearsures

In [32]:
n=0
M_buffalo=48000
g=48000/2
r_without_preventative_measures=2.1*10**(-6)

while logistic_non_recursive(n,r_without_preventative_measures,M_buffalo)<=g:
    n+=1

In [33]:
n

112

$Therefore\  n≥112$